src/KubernetesClient/Config.php

```diff
@@ -177,6 +177,10 @@ public static function shutdown()
      */
     public static function InClusterConfig()
     {
+        if (!file_exists('/var/run/secrets/kubernetes.io/serviceaccount/token')) {
+            throw new \Exception('Config based off running inside a cluster does not exist.');
```

Comment:
Given the finality of the situation here should we throw Error?

Reply:
Severity should be same as with problems during token fetching, it prevents the user from obtaining a running config. So I guess yes. But then it should as well be changed in `Config::BuildConfigFromFile()`, line 218, don't you think?

Reply:
Are you suggesting we try/catch the yaml functions and rethrow as Errors instead of Exceptions?

Reply:
I was not, but now I am =)
throwing Errors seems more appropriate. And then also catching and rethrowing the yamls exception.
But back to my earlier comment, I was suggesting throwing an Error too instead of that Exception which is thrown if the configfile does not exist.

Reply:
Yeah, go ahead and draft up the changes and I think we're in alignment :)

Reply:
Finally done. Sorry for the long wait =)

Reply:
Did we decide to keep throwing an Exception instead of an Error?

Reply:
No we did not, that was just my blindness =) but should be resolved now.